integrationtests/src/test/java/org/neo4j/procedure/ProcedureIT.java

```diff
@@ -362,6 +363,22 @@ public void shouldNotBeAbleToCallWriteProcedureThroughReadProcedure() throws Thr
     }
 
     @Test
+    public void shouldNotBeAbleToCallReadProcedureThroughWriteProcedureInWriteOnlyTransaction() throws Throwable
```

Comment:
It would be nice to have a positive version of this test, eg. `shouldCallReadProcedureThroughWriteProcedureInReadWriteTransaction`, so that we know that nesting procedures in an allowed way works.